packages/flutter/test/cupertino/date_picker_test.dart

```diff
@@ -1197,39 +1197,31 @@ void main() {
       }
 
       await tester.pumpWidget(buildApp(CupertinoDatePickerMode.time));
-      if (!skipPerspectiveTextGoldens) {
-        await expectLater(
-          find.byType(CupertinoDatePicker),
-          matchesGoldenFile('date_picker_test.time.initial.png'),
-        );
-      }
+      await expectFlakyGolden(
+        find.byType(CupertinoDatePicker),
+        'date_picker_test.time.initial.png',
```

Comment:
Oh right silly me. 🤦  I'll fix that.

```dart
if (perspectiveTestIsFlaky) {
  await expectFlakyGolden(
    find.byType(CupertinoDatePicker),
    'date_picker_test.time.initial.png',
  );
} else {
  await expectLater(
    find.byType(CupertinoDatePicker),
    matchesGoldenFile('date_picker_test.time.initial.png'),
  );
}
```

I am partial to this first case you provided. We can add expectFlakyGolden to the tech debt cost calculator and have easier follow up and maintenance. Ideally it would be nice to pass a list of platforms to filter for, but there isn't a clean way to include web with the target platform enum. I want to avoid making the soecial method have a non-flaky mode, so we don't end up with multiple types non-flaky calls `expectFlakyGolden(isFlaky = false)` versus `matchesGoldenFile`

Plus, the other with the flag required importing AsyncMatcher, which required `// ignore: implementation_imports`.

Reply:
SGTM. I imagine we won't have too many flakes happen all at once, so the value of code deduplication is limited. Making it easier to track tech debt is a good argument too.